### PR TITLE
Using httplib instead of httplib2 to get GCE metadata.

### LINKED ISSUE #575: '_implicit_environ.compute_engine_id()' may raise
I just saw this while running tests:

```
ERROR: test_set_explicit_None_wo_env_var_set (gcloud.datastore.test___init__.Test_set_default_dataset_id)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/datastore/test___init__.py", line 71, in test_set_explicit_None_wo_env_var_set
    self._callFUT(None)
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/datastore/test___init__.py", line 31, in _callFUT
    return set_default_dataset_id(dataset_id=dataset_id)
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/datastore/__init__.py", line 91, in set_default_dataset_id
    dataset_id = _implicit_environ.compute_engine_id()
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/datastore/_implicit_environ.py", line 73, in compute_engine_id
    response, content = http.request(uri, method='GET', headers=headers)
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/.tox/cover/lib/python2.7/site-packages/httplib2/__init__.py", line 1593, in request
    (response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/.tox/cover/lib/python2.7/site-packages/httplib2/__init__.py", line 1335, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/.tox/cover/lib/python2.7/site-packages/httplib2/__init__.py", line 1291, in _conn_request
    response = conn.getresponse()
  File "/opt/Python-2.7.9/lib/python2.7/httplib.py", line 1061, in getresponse
    raise ResponseNotReady()
```

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #575.

Feel free to run the bash script I [included][1] in the bug to torture test the fix.

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/issues/575#issuecomment-71943675

### CODE REVIEW

**Comment by dhermes** [comment]
I realized I need to use the `_Httplib` stub in the default fall through as well. Will fix when back at keyboard

**Comment by dhermes** [comment]
@tseaver PTAL, with 8dc13b522e0a0fbfa8c16e278720962b9d35bee5 I made sure all calls to `set_default_dataset_id` have the `httplib` stub.

File: gcloud/datastore/_implicit_environ.py [hunk 1]
```diff
@@ -64,17 +64,20 @@ def compute_engine_id():
...
 
-    response = content = None
+    content = None
     try:
-        response, content = http.request(uri, method='GET', headers=headers)
-    except socket.timeout:
+        connection.request('GET', uri_path, headers=headers)
+        response = connection.getresponse()
+        if response.status == 200:
+            content = response.read()
```
Comment (tseaver) [inline gcloud/datastore/_implicit_environ.py]: Could this be `return response.read()` (eliminating the need for the `content` variable)?

File: gcloud/datastore/test___init__.py [hunk 2]
```diff
@@ -30,12 +30,19 @@ def _callFUT(self, dataset_id=None):
...
         from gcloud._testing import _Monkey
+        from gcloud.datastore import _DATASET_ENV_VAR_NAME
+        from gcloud.datastore import _implicit_environ
+
         environ = {_DATASET_ENV_VAR_NAME: implicit_dataset_id}
-        return _Monkey(os, getenv=environ.get)
+        httplib = httplib or _Httplib(None, status=404)
+        return nested(_Monkey(os, getenv=environ.get),
+                      _Monkey(_implicit_environ, httplib=httplib,
+                              app_identity=app_identity))
```
Comment (tseaver) [inline gcloud/datastore/test___init__.py]: Cool -- I didn't know about `contextlib.nested`.

File: gcloud/datastore/_implicit_environ.py (hunk 1 above)
Comment (dhermes) [inline gcloud/datastore/_implicit_environ.py]: Yep. Will update.

File: gcloud/datastore/test___init__.py (hunk 2 above)
Comment (dhermes) [inline gcloud/datastore/test___init__.py]: Me neither, but I'm happy I didn't have to roll my own :)

**Comment by dhermes** [comment]
@tseaver Any other comments before I push the `content = None` fix?

**Comment by tseaver** [comment]
Nope -- "merge when ready, Gridley!"

**Pull request merged**